### PR TITLE
MAGN-10331 Create Revit 2018 DynamoRevit branch

### PR DESCRIPTION
**Pull request by sm6srw**
### Purpose

Make initial needed code changes for the new Revit 2018 branch and merge in 2017 changes added since the the 2018 branch was created. 

### Declarations

Check these if you believe they are true

- [X] The code base is in a better state after this PR
- [X] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [X] User facing strings, if any, are extracted into `*.resx` files
- [ ] Snapshot of UI changes, if any.

### Reviewers

@mjkkirschner @dobriai

### FYIs

@Benglin

### CODE REVIEW

File: src/Libraries/RevitNodes/Elements/ModelCurve.cs [hunk 1]
```diff
@@ -282,7 +282,7 @@ private static Curve Flatten3dCurveOnPlane(Curve c, Plane plane)
                     projPoints.Add(proj);
                 }
 
-                return Autodesk.Revit.DB.NurbSpline.Create(projPoints, ns.Weights.Cast<double>().ToList(), ns.Knots.Cast<double>().ToList(), ns.Degree, ns.isClosed, ns.isRational);
+                return Autodesk.Revit.DB.NurbSpline.CreateCurve(ns.Degree, ns.Knots.Cast<double>().ToList(), projPoints, ns.Weights.Cast<double>().ToList());
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/ModelCurve.cs]: does this method compute closed and rational properties itself?

**Comment by mjkkirschner** [comment]
@sm6srw looks fine to me, - have you been able to run any of the tests on the this branch yet?

File: src/Libraries/RevitNodes/Elements/ModelCurve.cs (hunk 1 above)
Comment (sm6srw) [inline src/Libraries/RevitNodes/Elements/ModelCurve.cs]: Yes

**Comment by sm6srw** [comment]
I have verified this manually against my own Revit build so far. But I think that a slightly modified RTF will work using a daily install build. I will try that tonight. I will then merge in the latest from the 2017 branch one more time before the final merge.

**Comment by sm6srw** [comment]
@mjkkirschner  I have been running some tests and it looks good so far. I will merge this shortly.

**Pull request merged**